src/react/hooks/paragon/useParagonThemeVariants.test.js

```diff
@@ -0,0 +1,159 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { getConfig } from '../../../config';
+import { logError } from '../../../logging';
+import useParagonThemeVariants from './useParagonThemeVariants';
+
+jest.mock('../../../logging');
+
+const mockAddEventListener = jest.fn();
+const mockRemoveEventListener = jest.fn();
+const mockOnChange = jest.fn();
+
+Object.defineProperty(window, 'matchMedia', {
+  value: jest.fn(() => ({
+    addEventListener: mockAddEventListener,
+    removeEventListener: mockRemoveEventListener,
+    onchange: mockOnChange,
+  })),
+});
+
+describe('useParagonThemeVariants', () => {
+  const themeOnLoad = jest.fn();
+  let initialState = false;
+
+  afterEach(() => {
+    document.head.innerHTML = '';
+    themeOnLoad.mockClear();
+    initialState = false;
+    logError.mockClear();
+    mockAddEventListener.mockClear();
+    mockRemoveEventListener.mockClear();
+  });
+
+  it('should create the links tags for each theme variant and change the state to true when are loaded', () => {
+    themeOnLoad.mockImplementation(() => { initialState = true; });
+    const themeVariants = {
+      light: {
+        urls: {
+          default: 'https://cdn.jsdelivr.net/npm/@edx/paragon@$paragonVersion/dist/light.min.css',
+          brandOverride: 'https://cdn.jsdelivr.net/npm/@edx/brand@$brandVersion/dist/light.min.css',
+
+        },
+      },
+      dark: {
+        urls: {
+          default: 'https://cdn.jsdelivr.net/npm/@edx/paragon@$paragonVersion/dist/dark.min.css',
+          brandOverride: 'https://cdn.jsdelivr.net/npm/@edx/brand@$brandVersion/dist/dark.min.css',
+
+        },
+      },
+    };
+    const currentThemeVariant = 'light';
+    renderHook(() => useParagonThemeVariants({ themeVariants, currentThemeVariant, onLoad: themeOnLoad }));
+    const themeLinks = document.head.querySelectorAll('link');
+    act(() => { themeLinks.forEach((link) => link.onload()); });
+
+    expect(themeLinks.length).toBe(4);
+    expect(initialState).toBeTruthy();
```

Comment:
Similar to other comment above, would recommend asserting that `themeOnLoad` was called, in place of the side of effect of what `themeOnLoad` mock is set up to do (i.e., mutate the `initialState`).